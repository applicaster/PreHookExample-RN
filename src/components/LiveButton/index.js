import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  Linking,
  StyleSheet,
  TouchableOpacity,
 } from 'react-native';
import hexToRgb from 'hex-to-rgb';

const buttonSize = 28;
const styles = StyleSheet.create({
  liveButton: {
    marginRight: 8,
    height: buttonSize,
    width: buttonSize,
  },
});

const LIVE_BUTTON_ASSET = 'live_button';

class LiveButton extends Component {
  constructor(props) {
    super(props);
    this.openLiveVideo = this.openLiveVideo.bind(this);
  }
  
  getButtonColor() {
    const { isLive } = this.props;
    const { textColor, mainColor } = this.context;
    const rgb = hexToRgb(textColor || '#FFFFFF');
    const offlineColor = `rgba(${rgb[0]},${rgb[1]},${rgb[2]}, 0.60)`;
    return isLive ? { tintColor: mainColor } : { tintColor: offlineColor };
  }

  openLiveVideo() {
    const { liveUrl } = this.props;
    Linking.openURL(liveUrl).catch(err => console.error('Unable to open live video', err));
  }
  
  renderActiveState() {
    const buttonColor = this.getButtonColor();
    return (<TouchableOpacity onPress={this.openLiveVideo}>
      <Image style={[styles.liveButton, buttonColor]} source={{ uri: LIVE_BUTTON_ASSET }} />
    </TouchableOpacity>);
  }

  renderInactiveState() {
    const buttonColor = this.getButtonColor();
    return <Image style={[styles.liveButton, buttonColor]} source={{ uri: LIVE_BUTTON_ASSET }} />;
  }

  render() {
    const { isLive } = this.props;
    return isLive ? this.renderActiveState() : this.renderInactiveState();
  }
}

LiveButton.propTypes = {
  isLive: PropTypes.bool,
  liveUrl: PropTypes.string,
};

LiveButton.contextTypes = {
  textColor: PropTypes.string,
  mainColor: PropTypes.string,
};

export default LiveButton;
