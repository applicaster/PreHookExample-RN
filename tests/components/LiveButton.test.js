import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import LiveButton from '../../src/components/LiveButton';

describe('LiveButton', () => {
  test('component renders correctly when it is not live', () => {
    const html = renderer.create(
      <LiveButton />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when it is live', () => {
    const html = renderer.create(
      <LiveButton isLive />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
