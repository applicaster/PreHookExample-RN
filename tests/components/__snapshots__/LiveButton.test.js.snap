// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveButton component renders correctly when it is live 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={
      Object {
        "uri": "live_button",
      }
    }
    style={
      Array [
        Object {
          "height": 28,
          "marginRight": 8,
          "width": 28,
        },
        Object {
          "tintColor": undefined,
        },
      ]
    }
  />
</View>
`;

exports[`LiveButton component renders correctly when it is not live 1`] = `
<Image
  source={
    Object {
      "uri": "live_button",
    }
  }
  style={
    Array [
      Object {
        "height": 28,
        "marginRight": 8,
        "width": 28,
      },
      Object {
        "tintColor": "rgba(255,255,255, 0.60)",
      },
    ]
  }
/>
`;
